src/Primitives/arduino.cpp

```diff
@@ -491,6 +508,36 @@ def_prim(chip_digital_write, twoToNoneU32) {
     return true;
 }
 
+def_prim_reverse(chip_digital_write) {
+    for (IOStateElement state : external_state) {
+        if (!state.output) {
+            continue;
+        }
+
+        if (state.key[0] == 'p') {
+            invoke_primitive(m, "chip_digital_write", stoi(state.key.substr(1)),
+                             (uint32_t)state.value);
+        }
+    }
+}
```

Comment:
Will this code not be the same for all `chip_digital_write` primitives, regardless of platform? Since you are just invoking the primitive.

(in fact this will the strategy for most primitives)

I think we can generalize this and move to a `reversibility` file to avoid code duplication. In will make implementing the reversibility for other primitives easier as well.

Reply:
Added as separate issue #263.